app/src/main/java/org/oppia/android/app/home/recentlyplayed/RecentlyPlayedFragmentPresenter.kt

```diff
@@ -111,39 +109,80 @@ class RecentlyPlayedFragmentPresenter @Inject constructor(
             itemList.add(ongoingStoryViewModel)
           }
         }
+
+        if (it.recommendedStoryList.suggestedStoryCount > 0) {
+          val showDivider = itemList.isNotEmpty()
+          val recommendedSectionTitleViewModel =
+            SectionTitleViewModel(
+              activity.getString(R.string.recommended_stories),
+              showDivider
+            )
+          itemList.add(recommendedSectionTitleViewModel)
+          for (suggestedStory in it.recommendedStoryList.suggestedStoryList) {
+            val ongoingStoryViewModel =
+              OngoingStoryViewModel(
+                suggestedStory,
+                entityType,
+                fragment as OngoingStoryClickListener
+              )
+            itemList.add(ongoingStoryViewModel)
+          }
+        }
+
         binding.ongoingStoryRecyclerView.layoutManager =
-          createLayoutManager(it.recentStoryCount, it.olderStoryCount)
+          createLayoutManager(
+            it.recommendedStoryList.recentlyPlayedStoryCount,
+            it.recommendedStoryList.olderPlayedStoryCount,
+            it.recommendedStoryList.suggestedStoryCount
+          )
         ongoingListAdapter.notifyDataSetChanged()
       }
     )
   }
 
-  private fun getAssumedSuccessfulOngoingStoryList(): LiveData<OngoingStoryList> {
+  private fun getAssumedSuccessfulOngoingStoryList(): LiveData<RecommendedActivityList> {
     // If there's an error loading the data, assume the default.
     return Transformations.map(ongoingStoryListSummaryResultLiveData) {
       it.getOrDefault(
-        OngoingStoryList.getDefaultInstance()
+        RecommendedActivityList.getDefaultInstance()
       )
     }
   }
 
   private fun createLayoutManager(
     recentStoryCount: Int,
-    oldStoryCount: Int
+    oldStoryCount: Int,
+    suggestedStoryCount: Int
   ): RecyclerView.LayoutManager {
     val sectionTitle0Position = if (recentStoryCount == 0) {
       // If recent story count is 0, that means that section title 0 will not be visible.
       -1
     } else {
       0
     }
-    val sectionTitle1Position = if (oldStoryCount == 0) {
-      // If old story count is 0, that means that section title 1 will not be visible.
-      -1
-    } else if (recentStoryCount == 0) {
-      0
-    } else {
-      recentStoryCount + 1
+    val sectionTitle1Position = when {
+      oldStoryCount == 0 -> {
+        // If old story count is 0, that means that section title 1 will not be visible.
+        -1
+      }
+      recentStoryCount == 0 -> {
+        0
+      }
+      else -> {
```

Comment:
Can we omit the parenthesis here? Just a better readability
```
 val sectionTitle1Position = when {
      // If old story count is 0, that means that section title 1 will not be visible.	      
      oldStoryCount == 0 -> -1
      recentStoryCount == 0 -> 0
      else -> recentStoryCount + 1
    }
```

Reply:
Done.